docs/system/charge_controller.md

```diff
@@ -1,9 +1,94 @@
+---
+sidebarDepth: 2
+---
+
 # Charge Controller
 
+A charge controller regulates the voltage and/or current flowing into batteries. By doing so, it prevents the batteries from overcharging and ensures good battery lifetime.
+
+## Solar
+
+Solar energy is the predominant method for off-grid electricity generation, as it does not require any moveable parts and is easily installed.
+PWM (Pulse Width Modulation) charge controller is in pratice a switch connected between solar panel and a battery. During bulk or constant current load, the switch is simply closed and the battery is charged with whatever current the panel can provide. The panel's voltage is pulled down which effectively lowers the panels efficiency since it is not operating in its **maximum power point**. After bulk charging, the pwm-controller will apply different duty cycles resulting in current boosts, effectively applying a constant voltage. The cycles are adapted depending on the the battery's charge state. The **interactive** example in Figure [1] shows the power effectively used for charging a battery with a given base voltage. The closer the rated battery voltage to the maximum power point voltage, the higher the overall efficiency.
```

Comment:
```suggestion
PWM (Pulse Width Modulation) charge controller is in practice a switch connected between solar panel and a battery. During bulk or constant current charging phase, the switch is simply closed and the battery is charged with whatever current the panel can provide. The panel's voltage is pulled down which effectively lowers the panels efficiency since it is not operating in its **maximum power point**. After bulk charging, the PWM charge controller will apply a defined duty cycle interrupting the current for short periods of time, which results in a constant average voltage. The cycles are adapted depending on the the battery's charge state. The **interactive** example in Figure 1 shows the power effectively used for charging a battery with a given base voltage. The closer the rated battery voltage to the maximum power point voltage, the higher the overall efficiency.
```